### PR TITLE
add drop down menu for showing main account and sub account

### PR DESCRIPTION
**Pull request by liyaqin1**

### CODE REVIEW

**Comment by liyaqin1** [comment]
Tested on dev1.
If I select `Main Account`, it shows the main account hosts; if I select `Sub Account`, it shows the sub account host which in this case means nothing.
Screenshot 2023-08-23 at 1 13 24 PM
Screenshot 2023-08-23 at 1 13 36 PM

File: deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl [hunk 1]
```diff
@@ -13,29 +13,69 @@
...
-                           {% endif %}
-                           <i class="fa fa-fw {{ agentStat | agentIcon }}"></i>
+                        type="button" class="deployToolTip btn btn-xs {{ agentStat | agentButton }} host-btn"
+                        title="{{ agentStat | agentTip }}">
+                        {% if report.showMode != "compact" %}
+                        <small>{{ agentStat.agent.hostName }}</small>
+                        {% endif %}
+                        <i class="fa fa-fw {{ agentStat | agentIcon }}"></i>
+                        </a>
+                        {% elif report.showMode == "main" %}
```
Comment (ntascii) [inline deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl:25]: nit:mainAcc and subAcc

File: deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl [hunk 2]
```diff
@@ -13,29 +13,69 @@
...
+                        {% if not agentStat.agent.accountId or agentStat.agent.accountId == 998131032990 %}
```
Comment (ntascii) [inline deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl:26]: let's have a primaryAccount flag for each agent in py code.

File: deploy-board/deploy_board/webapp/agent_report.py [hunk 3]
```diff
@@ -148,6 +148,7 @@ def gen_report(request, env, progress, sortByStatus="false"):
     deployStats[env['deployId']] = currentDeployStat
 
     for agent in progress["agents"]:
+        agent["accountId"] = hosts_helper.get_accout_id_by_name(request, agent["hostName"])
```
Comment (ntascii) [inline deploy-board/deploy_board/webapp/agent_report.py:151]: do we have host_id in agent? some hosts might not have hostName

File: deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl (hunk 1 above)
Comment (liyaqin1) [reply deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl:25]: sounds good

File: deploy-board/deploy_board/webapp/agent_report.py (hunk 3 above)
Comment (liyaqin1) [reply deploy-board/deploy_board/webapp/agent_report.py:151]: yes, we have host_id in agents table

File: deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl (hunk 2 above)
Comment (liyaqin1) [reply deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl:26]: sounds reasonable, thanks

File: deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl [hunk 4]
```diff
@@ -13,29 +13,69 @@
...
+               {% if agentStat.agent.hostId == 'i-0ce811ae17e6c1b78' %}
+               <a href="/env/{{env.envName}}/{{env.stageName}}/host/{{ agentStat.agent.hostName }}"
+               type="button" class="deployToolTip btn btn-xs {{ agentStat | agentButton }} host-btn"
+               title="{{ agentStat | agentTip }}">              
+               <small>{{ agentStat.agent.hostName }}</small>
+               <i class="fa fa-fw {{ agentStat | agentIcon }}"></i>
+               </a>
+               {% endif %}
+               {% elif report.showMode == "subAcct" %}
+               {% if accountIds['i-0a6485b3eed813298'] == 998131032990 %}
```
Comment (ntascii) [inline deploy-board/deploy_board/templates/deploys/deploy_progress.tmpl:70]: seems like for testing purpose? can we correct it

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Agents.java [hunk 5]
```diff
@@ -89,4 +95,23 @@ public long getCountByEnvName(
...
+    @Path("/env/{envId : [a-zA-Z0-9\\-_]+}/accountIds")
+    @ApiOperation(
+            value = "Get account id for a specific environment object",
+            notes = "Returns a mapping object of host id and account id given an environment id",
+            response = String.class, responseContainer = "Map")
+    public Map<String, String> getAccountIds(
+            @ApiParam(value = "Environment id", required = true)@PathParam("envId") String envId) throws Exception {
+        Map<String, String> result = new HashMap<String, String>();
+        List<AgentBean> agents = agentDAO.getAllByEnv(envId);
+        for (int i = 0; i < agents.size(); i++)
```
Comment (ntascii) [inline deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Agents.java:109]: for (AgentBean agent: agents)?

Comment (liyaqin1) [reply deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Agents.java:109]: sure

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Agents.java [hunk 6]
```diff
@@ -89,4 +95,23 @@ public long getCountByEnvName(
...
+            value = "Get account id for a specific environment object",
+            notes = "Returns a mapping object of host id and account id given an environment id",
+            response = String.class, responseContainer = "Map")
+    public Map<String, String> getAccountIds(
+            @ApiParam(value = "Environment id", required = true)@PathParam("envId") String envId) throws Exception {
+        Map<String, String> result = new HashMap<String, String>();
+        List<AgentBean> agents = agentDAO.getAllByEnv(envId);
+        for (int i = 0; i < agents.size(); i++) 
+        {
+            String hostId = agents.get(i).getHost_id();
```
[the remark this answers is not in the document]
Comment (liyaqin1) [reply deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Agents.java:111]: yea, yea, really make sense

**Review by ntascii — APPROVED** [review]
lgtm

**Pull request merged**